plugins/docker/pull.ml

```diff
@@ -17,19 +32,47 @@ module Value = Image
 
 let id = "docker-pull"
 
+let get_digest_from_manifest manifest arch =
+  let jstring = function `String s -> s |_ ->raise Not_found in
+  let jlist = function `List l -> l |_->raise Not_found in
+  let jassoc = function `Assoc x -> x | _->raise Not_found in
```

Comment:
There are convenience functions for these in yojson, e.g. `Yojson.Basic.Util.to_string`